engine/src/main/java/org/pentaho/di/job/JobMeta.java

```diff
@@ -436,8 +438,11 @@ public Object realClone( boolean doClear ) {
         // cloneNode is *probably* overkill
         jobMeta.localSharedObjects.saveSharedObject( dbType, entry.getKey(), entry.getValue().cloneNode( true ) );
       }
-      for ( SlaveServer slave : slaveServers ) {
-        jobMeta.getSlaveServers().add( (SlaveServer) slave.clone() );
+      //SlaveServers
+      String slaveServerType = SharedObjectsIO.SharedObjectType.SLAVESERVER.getName();
+      for ( Map.Entry<String, Node> entry : localSharedObjects.getSharedObjects( slaveServerType ).entrySet() ) {
+        // cloneNode is *probably* overkill
+        jobMeta.localSharedObjects.saveSharedObject( slaveServerType, entry.getKey(), entry.getValue().cloneNode( true ) );
       }
```

Comment:
This feels really verbose, like we're coupling all the meta classes with the shared object implementation even more tightly than it was before.